specs/docs/security/DataHubSecurity.md

```diff
@@ -0,0 +1,267 @@
+```javascript
+{
+  "local-name" : "updateJob",
+  "document-uri" : "/data-hub/5/impl/jobs.sjs",
+  "modules-database" : "%%mlModulesDbName%%",
+  "role" : [ "data-hub-job-internal" ]
+}
+```
+
+In JavaScript, in addition to the amp configuration file, you need to make use of the `module.amp` function to wrap your function on export. See [Amps and the module.amp Function](https://docs.marklogic.com/guide/jsref/functions#id_13020) 
+
+#### Example JavaScript amp using module.amp
+```javascript
+module.exports.updateJob = module.amp(
+  function updateJob(datahub, jobId, status, flow, step, lastCompleted, stepResponse) {
+  //...
+  }
+);
+```
+### When To Grant Privileges To User Roles And When To Use Amps
+
+User roles should be assigned privileges when the privilege is sufficiently restrictive and does not allow arbitrary code execution.
+
+If the privilege would allow arbitrary code execution, that is a sure indication that an amp should be used. For other privileges, consider if the privilege can be used beyond the intended purpose. Where possible use granular privileges (See [Granular Privileges](https://docs.marklogic.com/guide/security/granular)). If the user role can't use granular privileges and the required privilege would be too broad, (1) a new narrow execute privilege should be created, (2) the function that requires the privilege should check for the narrow privilege, and (3) the function should be amped with the broader privilege.
+
+Roles to be used with amps should be suffixed with `-internal` to indicate that the role is used for amps and NOT intended to be assigned to users.
+
+## Data Hub Roles and Privileges
+
+### data-hub-common
+
+Data Hub role containing the common privileges across Data Hub roles.
+
+#### Inherited Roles
+ - data-hub-module-reader
+ - rest-reader
+
+#### Privileges
+
+Privilege | Notes
+---------|----------
+ xdmp-eval-in | Required to specify database for REST Client. (Note: This privilege alone doesn't allow eval to be called. The REST API code does the neccesary amping for eval execution)
+ xdbc-eval-in | Same as xdmp-eval-in
+
+### data-hub-common-writer
+
+Data Hub role containing the common privileges across Data Hub writer roles.
+
+#### Inherited Roles
+ - data-hub-common
+ - rest-writer
+
+#### Privileges
+
+Privilege | Notes
+---------|----------
+ ps-user | Needed to record provenance data
+ xdmp-document-load | Needed to load data directly in the database
+ any-uri | Required to write documents to any URI without explicit privileges granted
+ any-collection | Required to write documents to any collection without explicit privileges granted
+
+### hub-central-user
+
+All Data Hub roles intended for Hub Central users inherit this role.
+
+#### Associate Authorities
+ - canLoginToHubCentral
+
+#### Inherited Roles
+ - data-hub-common
+
+### hub-central-downloader
+
+Role to determine if a user can download the Hub Central project.
+
+#### Associate Authorities
+ - canDownloadProject
```

Comment:
I think "canDownloadUserArtifacts" may be better here, since the entire project cannot be downloaded, just what we internally call "user artifacts".

Reply:
Also, I don't know yet if this role should have all the read-access roles for user artifacts, or if we'd used an amp in the DS endpoint to grab all of them. The latter is certainly possible - I did a little prototype of using xdmp.zip. So a DS endpoint that requires a certain privilege and then uses an amp to construct the zip is possible. But I don't know if that's worth calling out here - it would seem a little early to do that.